src/libraries/Pool.sol

```diff
@@ -111,23 +98,28 @@ library Pool {
         internal
         returns (int24 tick)
     {
-        if (self.slot0.sqrtPriceX96 != 0) revert PoolAlreadyInitialized();
+        if (self.slot0.sqrtPriceX96() != 0) revert PoolAlreadyInitialized();
 
         tick = TickMath.getTickAtSqrtPrice(sqrtPriceX96);
 
-        self.slot0 = Slot0({sqrtPriceX96: sqrtPriceX96, tick: tick, protocolFee: protocolFee, lpFee: lpFee});
+        self.slot0 = 
+            Slot0.wrap(bytes32(0))
+                .setSqrtPriceX96(sqrtPriceX96)
+                .setTick(tick)
+                .setProtocolFee(protocolFee)
+                .setLpFee(lpFee);
```

Comment:
How about a `pack` method if putting all components together? Should be more efficient.

Reply:
I'm afraid that this will lead to a somewhat significant increase in bytecode size for the sake of a tiny gain in a function that is called once. In my opinion, it's better to have smaller bytecode, but more aggressive optimizer settings

Reply:
I'll check what it does anyway

Reply:
@shuhuiluo I tried to implement it as:

```solidity
  function pack(uint160 _sqrtPriceX96, int24 _tick, uint24 _protocolFee, uint24 _lpFee) internal pure returns (Slot0 _result) {
        /// @solidity memory-safe-assembly
        assembly {
          _result := and(UINT160_MASK, _sqrtPriceX96)
          _result := or(_result, shl(TICK_OFFSET, and(INT24_MASK, _tick)))
          _result := or(_result, shl(PROTOCOL_FEE_OFFSET, and(UINT24_MASK, _protocolFee)))
          _result := or(_result, shl(LP_FEE_OFFSET, and(UINT24_MASK, _lpFee)))
      }      
  }
```

and I see no changes in snapshots at all...

Reply:
Interesting. The compiler may be able to reduce the math and the masking on `0`. Still, appreciate it.